src/language-server/__tests__/document.test.ts

```diff
@@ -119,6 +174,101 @@ describe("extractGraphQLDocuments", () => {
       expect(documents?.length).toEqual(1);
       expect(documents?.[0].syntaxErrors.length).toBe(0);
       expect(documents?.[0].ast?.definitions.length).toBe(2);
+      expect(replaceWhitespaceWithDots(documents![0].source.body))
+        .toMatchInlineSnapshot(`
+"
+········{
+··········hero·{
+············...Hero_character
+··········}
+········}
+
+··································
+
+········{
+··········reviews(episode:·NEWHOPE)·{
+············...ReviewList_reviews
+··········}
+········}
+
+····································
+····"
+`);
+    });
+
+    test("fixtures", () => {
+      function loadFixture(name: string) {
+        const path = join(__dirname, "fixtures", "documents", name);
+        const body = readFileSync(path, "utf8");
+
+        return extractGraphQLDocuments(
+          TextDocument.create(`file://${path}.js`, "javascript", 1, body),
+        )!;
+      }
+      function documentName(document: GraphQLDocument) {
+        expect(document.syntaxErrors.length).toBe(0);
+        let first = document.ast?.definitions[0];
+        expect(first).toBeDefined();
+        expect(first!.kind).toBe("OperationDefinition");
+        first = first as OperationDefinitionNode;
+        return (first.name && first.name.value) || "Unnamed";
+      }
+
+      expect(loadFixture("commentWithTemplate.ts").map(documentName))
+        .toMatchInlineSnapshot(`
+Array [
+  "Q1",
+  "Q2",
+  "Q3",
+  "Q4",
+  "Q6",
+  "Q8",
+  "Q9",
+  "Q10",
```

Comment:
We just test that all queries in the fixture are found by looking at the name, and the `documentName` function also checks that there were no syntax errors in the document.